tests/test_input.py

```diff
@@ -576,6 +588,83 @@ def test_config_parsing_bad_objective():
+    """
+
+    train_dir = join(_my_dir, 'train')
+    test_dir = join(_my_dir, 'test')
+    output_dir = join(_my_dir, 'output')
+
+    # make a simple config file that has a bad task
+    # but everything else is correct
+    values_to_fill_dict = {'experiment_name': 'config_parsing',
+                           'task': 'evaluate',
+                           'train_directory': train_dir,
+                           'test_directory': test_dir,
+                           'featuresets': "[['f1', 'f2', 'f3']]",
+                           'learners': "['LogisticRegression']",
+                           'log': output_dir,
+                           'results': output_dir,
+                           'objective': "['accuracy']"}
+
+    config_template_path = join(_my_dir, 'configs',
+                                'test_config_parsing.template.cfg')
+    config_path = fill_in_config_options(config_template_path,
+                                         values_to_fill_dict,
+                                         'bad_objective')
+
+    yield check_config_parsing_type_error, config_path
+
+
+def test_config_parsing_bad_objectives():
+    """
+    Test to ensure config file parsing raises an error with a grid objectives given as a string
+    """
+
+    train_dir = join(_my_dir, 'train')
+    test_dir = join(_my_dir, 'test')
+    output_dir = join(_my_dir, 'output')
+
+    # make a simple config file that has a bad task
+    # but everything else is correct
+
+    config_template_path = join(_my_dir, 'configs',
+                                'test_objectives_error.template.cfg')
+    values_to_fill_dict = {'train_directory': train_dir,
+                           'log': output_dir,
+                           'results': output_dir}
+    config_path = fill_in_config_options(config_template_path,
+                                         values_to_fill_dict,
+                                         'bad_objectives')
+    yield check_config_parsing_type_error, config_path
+
+
+@raises(ValueError)
+def test_config_parsing_bad_objective_and_objectives():
+    """
+    Test to ensure config file parsing raises an error with 
+    a grid objectives and objective both given non default values
+    """
+
+    train_dir = join(_my_dir, 'train')
+    test_dir = join(_my_dir, 'test')
+    output_dir = join(_my_dir, 'output')
+
+    # make a simple config file that has a bad task
+    # but everything else is correct
+
+    config_template_path = join(_my_dir, 'configs',
+                                'test_objective_and_objectives_conflict_error.template.cfg')
+    values_to_fill_dict = {'train_directory': train_dir,
+                           'log': output_dir,
+                           'results': output_dir}
+    fill_in_config_options(config_template_path,
+                           values_to_fill_dict,
+                           'bad_objective_and_objectives')
+
```

Comment:
Again, I am not sure why you are using a template pre-filled with bad values instead of using the regular config parsing template and passing in the values as part of `values_to_fill_dict`? And this way, you don't need the decorator right?